src/pulsar_producer.erl

```diff
@@ -85,12 +87,15 @@
 -define(MAX_REQ_ID, 4294836225).
 -define(MAX_SEQ_ID, 18445618199572250625).
 
+-define(DEFAULT_MAX_INFLIGHT, 1000).
```

Comment:
maybe easier to understand and use if this is to control the max number of in-flight batches, instead of max inflight of calls ?
and by default 100 should be good.
like max_send_ahead for wolff_producer

Reply:
it seems to already be the number of inflight messages, despite the name

Reply:
emqx uses default value max_inflight = 10 in the schema

Reply:
... as it counts individual messages, not calls:

https://github.com/emqx/pulsar-client-erl/blob/1865307483657da059a5311d87190f630c3acb22/src/pulsar_producer.erl#L825-L830

Reply:
and I tested with 10, and it lead to a lot of batching, hurting throughput

Reply:
... though, I just tested again, and it worked fine with 10 :thinking: